app/frontend/src/components/Datepicker.tsx

```diff
@@ -53,10 +52,8 @@ export default function Datepicker({
             label={label}
             minDate={minDate}
             onChange={(date) => {
-              //user might be typing, so check the date is valid before doing conversions
-              date?.isValid()
-                ? onChange(date?.format().split("T")[0])
-                : onChange(date);
+              // user might be typing, so check the date is valid before doing conversions
+              date?.isValid() ? onChange(date.toDate()) : onChange(date);
```

Comment:
This was a hacky thing for the string value. I wonder if `onChange={onChange}` will work now it's all `Date`s?